src/socket-client.ts

```diff
@@ -177,18 +179,33 @@ export class SocketClient extends EventEmitter {
             connectOptions.upgrade = false;
         }
 
-        /**
-         * Identify this endpoint to the endpoint for session-to-socket mapping.
-         * Without this, the backend cannot "take first steps" because it can't
-         * reach the client!
-         */
-        connectOptions["query"] = {
-            sessionId: encodeURIComponent(this.socketOptions.sessionId),
-            urlToken: encodeURIComponent(this.socketURLToken),
-            userId: encodeURIComponent(this.socketOptions.userId),
+        if (!this.socketOptions.enableInnerSocketHandshake) {
+            /**
+             * Identify this endpoint to the endpoint for session-to-socket mapping.
+             * Without this, the backend cannot "take first steps" because it can't
+             * reach the client!
+             */
+            connectOptions["query"] = {
+                sessionId: encodeURIComponent(this.socketOptions.sessionId),
+                urlToken: encodeURIComponent(this.socketURLToken),
+                userId: encodeURIComponent(this.socketOptions.userId),
+            }
+        } else {
+            /**
+             * Instructs the backend to emit a "handshake" event
+             * as soon as the connection is established.
+             * The client will answer that handshake in a handler
+             * set up below.
+             * 
+             * This way, userId, sessionId and urlToken are not visible
+             * to an interceptor!
+             */
+            connectOptions["query"] = {
+                handshake: "true"
```

Comment:
not super happy with the naming since the object notation being socket.handshake.query.handshake in the code is more confusing than clear. especially if our own "handshake" is single purpose

Reply:
This is something that will be set implicitly by the library. I can add an explanatory comment indicating what this is used for.